app/services/data/get-caseload-progress.js

```diff
@@ -1,15 +1,14 @@
 const knex = require('../../../knex').web
 const orgUnitFinder = require('../helpers/org-unit-finder')
 const ORGANISATION_UNIT = require('../../constants/organisation-unit')
-const Promise = require('bluebird').Promise
 
 module.exports = function (id, type) {
   var orgUnit = orgUnitFinder('name', type)
   var table = orgUnit.caseProgressView
-  var whereObject = {}
+  var whereString = ''
 
-  if (id !== undefined) {
-    whereObject.id = id
+  if (id !== undefined && (!isNaN(parseInt(id, 10)))) {
+    whereString += ' WHERE id = ' + id
```

Comment:
this should be `whereString =` not needed to do `+=`

Reply:
I'll change this now.